### PR TITLE
Add Fomu Hacker platform.

### PR DESCRIPTION
**Pull request by rroohhh**
fyi: @mithro

### CODE REVIEW

File: nmigen_boards/fomu_hacker.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            mosi="F1",
+            miso="E1",
+            attrs=Attrs(IO_STANDARD="SB_LVCMOS33"),
+        ),
+    ]
+
+    connectors = [Connector("touch_pins", 0, "F4 E5 E4 F2")]
+
+    def toolchain_program(self, products, name):
+        iceprog = os.environ.get("DFU_UTIL", "dfu-util")
```
Comment (whitequark) [inline nmigen_boards/fomu_hacker.py]: Wrong name.

File: nmigen_boards/fomu_hacker.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        *SPIFlashResources(0,
+            cs="C1",
+            clk="D1",
+            mosi="F1",
+            miso="E1",
+            attrs=Attrs(IO_STANDARD="SB_LVCMOS33"),
+        ),
+    ]
+
+    connectors = [Connector("touch_pins", 0, "F4 E5 E4 F2")]
```
Comment (whitequark) [inline nmigen_boards/fomu_hacker.py]: Why is this a connector?

**Review by whitequark — CHANGES_REQUESTED** [review]

**Comment by whitequark** [comment]
Also, is the "Fomu Hacker" different from normal "Fomu"? /cc @mithro

File: nmigen_boards/fomu_hacker.py (hunk 2 above)
Comment (anuejn) [reply nmigen_boards/fomu_hacker.py]: This is due to the possibility to use those as gpio pins. It is a kind of weird design ([see schematic](https://github.com/im-tomu/fomu-hardware/blob/master/hacker/releases/v0.0-19-g154fecc/schematic.pdf)). According to @xobs, they can be used as resistive touch sensors (with internal pulldown and a 3v3 plane on top of the board) but it seems to be fairly untested.

Comment (whitequark) [reply nmigen_boards/fomu_hacker.py]: I don't see them connected anywhere on schematic?

**Comment by anuejn** [comment]
See the [the fomu tutorial section hardware](https://github.com/im-tomu/fomu-workshop#required-hardware).
TL;DR: yes

File: nmigen_boards/fomu_hacker.py (hunk 2 above)
Comment (anuejn) [reply nmigen_boards/fomu_hacker.py]: Yes, they are just exposed pads on the PCB

Comment (whitequark) [reply nmigen_boards/fomu_hacker.py]: OK. It sounds like they should really be 4 connectors, `Connector("pin", 0, "XX")` and so on.

**Comment by whitequark** [comment]
Thanks.

**Comment by mithro** [comment]
Just FYI There are currently 3 Fomu variants;
 a) Fomu EVT board (RPi Hat board used during development)
 b) Fomu Hacker board (Cheap free board from @mithro)
 c) Fomu PVT board (Production board from @xobs)
